### DISCUSSION TITLE
All SSR Apollo examples are gone?

### DISCUSSION THREAD

**Discussion by leocavalcante** [vercel/next.js#14856] [Help]
It seams that all Apollo examples was moved to SSG 😢. There was one left showing how to make the plain-old SSR using `@apollo/react-ssr`?

**Comment by rafaelalmeidatk** [comment]
Hi, you can change the usage of `getStaticProps` to `getServerSideProps` and it will be SSR!

**Reply by leocavalcante** [reply]
I don't think it is that simple because I have to know [what queries my components ](https://github.com/vercel/next.js/blob/canary/examples/with-typescript-graphql/pages/index.tsx#L6)are doing then [repeat them all at `getServerSideProps` ](https://github.com/vercel/next.js/blob/canary/examples/with-typescript-graphql/pages/index.tsx#L24)in order to pass the cache result as [`initialApolloState`](https://github.com/vercel/next.js/blob/canary/examples/with-typescript-graphql/pages/index.tsx#L29). It is basically reimplementing my own `getDataFromTree` instead of using the one from `@apollo/react-ssr` like before.

**Reply by rafaelalmeidatk** [reply]
This is no different from the SSG examples tho. `getDataFromTree` has been removed from the examples because traversing the element tree twice is way slower than doing the requests directly. You might want to give a look at https://github.com/hashicorp/rivet-graphql, it uses fragments to make it easier to keep track of the queries the components in the page need

If you really want to use `getDataFromTree`, you can still check the old examples with the File History, but I am not sure we will have updated examples using it for the mentioned reasons

**Reply by leocavalcante** [reply]
I didn't know about the performance issue, thanks for the heads up, I'll look for alternatives like the mentioned.

**Reply by Herlevsen** [reply]
Of course it is slower, by nature, but does it matter? Probably not 99.9% of the time. Just seems like premature optimization at the cost of maintainability (IMO). I don't find the current examples to be optimal, since the Apollo docs pushes hooks as the standard way of executing queries and mutations. That is going to be very confusing for beginners.
